golang/pkg/dagent/dagent.go

```diff
@@ -50,6 +51,13 @@ func Serve(cfg *config.Configuration) {
 func grpcClose(ctx context.Context, reason agent.CloseReason) error {
 	if reason == agent.CloseReason_SELF_DESTRUCT {
 		return update.RemoveSelf(ctx)
+	} else if reason == agent.CloseReason_SHUTDOWN {
+		log.Info().Msg("Remote shutdown requested")
+		err := utils.StopSelf(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Docker failed to stop")
+			os.Exit(0)
```

Comment:
Stopping should be done in all environment using Exit 0, is there an issue with it?

Reply:
Docker restarts the container for some reason, but `ContainerStop` keeps it stopped.

Reply:
It is the `unless-stopped` restart policy's behavior, `on-failure` is the one we are looking for if we want 'exit 0' to be a valid termination state for dagent.

Reply:
I see, I'll change the install scripts then, and I'll be able to implement the Kubernetes exit as well I think.